From cf1524c614909172d04482f8eeef9d9f34223f36 Mon Sep 17 00:00:00 2001
From: sjayanti <satya.charitardha.jayanti@intel.com>
Date: Mon, 23 Jul 2012 18:51:29 +0530
Subject: [PORT FROM MAIN] Audio-MW: [ET][R3] The video recorder tone sounds too loudly via wired headset when tapping the record icon.

BZ: 33378

When Headset is connected and stream is ENFORCED_AUDIBLE for video recorder tone, then to avoid
sound level burst in user's ears, volume is attenuated by 6dB with this fix.

Change-Id: Id74a6a9c69820efc5bbdc8718c3e0af6baa51d8b
Orig-Change-Id: Ie9280601bcdb084f63d440b2b82e59240e9697fc
Signed-off-by: sjayanti <satya.charitardha.jayanti@intel.com>
Reviewed-on: http://android.intel.com:8080/81075
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 audio/AudioPolicyManagerBase.cpp |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 52d89ab..77692cf 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -2801,6 +2801,14 @@ float AudioPolicyManagerBase::computeVolume(int stream,
 
     volume = volIndexToAmpl(device, streamDesc, index);
 
+    // if a headset is connected and stream is ENFORCED_AUDIBLE (Camera shutter sound),
+    // then to avoid sound level burst in user's ears, attenuate the volume by 6dB.
+    if ((device &
+        (AudioSystem::DEVICE_OUT_WIRED_HEADSET |
+        AudioSystem::DEVICE_OUT_WIRED_HEADPHONE)) &&
+        (stream == AudioSystem::ENFORCED_AUDIBLE)) {
+        volume *= SONIFICATION_HEADSET_VOLUME_FACTOR;
+    }
     // if a headset is connected, apply the following rules to ring tones and notifications
     // to avoid sound level bursts in user's ears:
     // - always attenuate ring tones and notifications volume by 6dB
-- 
1.7.4.1

